src/app/item-review/item-review.component.js

```diff
@@ -209,6 +232,14 @@ function ItemReviewController(dimDestinyTrackerService, $scope, $rootScope) {
     }
   };
 
+  vm.translateReviewMode = function(review) {
+    if (!vm.defs || !vm.defs.ActivityMode) {
+      getDefinitions().then((defs) => { translateReviewMode(defs, review); });
+    }
+
+    return translateReviewMode(vm.defs, review);
+  };
```

Comment:
You should just always use the promise version. It'll return instantly if it's already resolved.

Reply:
`"[$rootScope:infdig] 10 $digest() iterations reached. Aborting!
Watchers fired in the last 5 iterations: []
http://errors.angularjs.org/1.6.9/$rootScope/infdig?p0=10&p1=%5B%5D"`

I must be doing something wrong in there?

Reply:
Oh, I see what's happening. You should never call a function from ng-bind or other watched expressions. Use a filter instead, or preprocess these values.

Reply:
Ah, that makes sense to me.

It’s using the same array of values processed out of the manifest, but that only needs to happen once. 

Is there an expected place to hold (lazy loaded) global state? I don’t know if we have any singletons in the app.